docs/DeveloperGuide.md

```diff
@@ -1,37 +1,351 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- This project was inspired by [AddressBook-Level3 (AB3)](https://github.com/se-edu/addressbook-level3), and several code structures were adapted for the MediTask application.
+- JSON serialization and deserialization functionality uses the [Jackson library](https://github.com/FasterXML/jackson).
+- Logging setup was referenced from open-source Java projects using `Logger`.
+
+---
+
+## Setting up, getting started
+Refer to the guide for 'Setting up and getting started.'
+
+---
+
+## Design
+
+### Architecture
+
+![Architecture](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the MediTask App.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main Components**
+
+`MediTask` is in charge of the CLI to launch and the exit.
+
+- At CLI launch, it initializes the other components in the correct sequence, and connects them up with each other.
+- At exit, it shuts down the other components and invokes cleanup methods where necessary.
+
+The bulk of the CLI's work id done by the following components:
+
+- **UI**: Handles the user interface and user interactions.
+- **Parser**: Interprets user commands and executes them.
+- **Command**: Contains the logic for executing user commands.
+- **StateManager**: Manages the state logic of the application and keeps track of the current state of the application.
+- **Storage**: Handles reading and writing data from and to the hard disk.
+- **Data**:  Holds the data of the App in memory.
+- **Common**: Contains common classes used by other components.
+
+### UI
+
+### Parser
+The Parser interface uses a series of classes to implement the various commands.
+![Parser_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParserClass.png)
+
+### Command
+The `commands` package includes the command pattern used in the application to handle user operations. An abstract `Command` class serves as the base for all specific command implementations. Each command class extends `Command` and overrides the `execute()` method to perform its intended action.
+
+![Command_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CommandClass.png)
+
+### Data
+
+### task
+The `task` package manages all task-related functionality. A `Task` class serves as the base for other task types: `Todo`, `Deadline`, and `Repeat`. Each task type extends `Task` and introduces additional attributes relevant to its behavior. The `TaskList` class maintains a collection of tasks and provides methods to add, delete, find and track the completion rate of tasks.
+
+![Task_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/TaskClassDiagram.png)
```

Comment:
The task class diagram format could be more appropriate for the class diagrams above